core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DatalakeV4Controller.java

```diff
@@ -28,11 +31,46 @@ public class DatalakeV4Controller implements DatalakeV4Endpoint {
     @Inject
     private WorkspaceService workspaceService;
 
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
     @Override
+    @InternalOnly
     public StackViewV4Responses list(@Deprecated String environment, @TenantAwareParam String environmentCrn) {
         if (StringUtils.isBlank(environmentCrn) || StringUtils.isNotBlank(environment)) {
             throw new BadRequestException("environment param is deprecated, only environmentCrn should be filled.");
         }
         return stackOperations.listByEnvironmentCrn(workspaceService.getForCurrentUser().getId(), environmentCrn, List.of(StackType.DATALAKE));
     }
+
+    @Override
+    public Boolean isDiskTypeChangeSupported(String platform) {
+        if (StringUtils.isBlank(platform)) {
+            throw new BadRequestException("Platform param is null or empty.");
+        }
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
+    }
+
+    @Override
+    public void updateDiskTypeAndSize(SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception {
+        DiskUpdateRequest diskUpdateRequest = sdxDiskUpdateRequest.getDiskUpdateRequest();
+        List<VolumeSetAttributes.Volume> volumesToUpdate = sdxDiskUpdateRequest.getVolumesToUpdate();
+        String cloudPlatform = sdxDiskUpdateRequest.getCloudPlatform();
+        diskUpdateService.updateDiskTypeAndSize(diskUpdateRequest, volumesToUpdate, cloudPlatform);
+    }
+
+    @Override
+    public void stopCluster(long stackId) throws Exception {
+        diskUpdateService.stopCluster(stackId);
+    }
+
+    @Override
+    public void startCluster(long stackId) throws Exception {
+        diskUpdateService.startCluster(stackId);
+    }
+
+    @Override
+    public void resizeDisks(long stackId, String instanceGroup) throws Exception {
+        diskUpdateService.resizeDisks(stackId, instanceGroup);
+    }
```

Comment:
none of these will work, since there is no parameter for these which can help our authz framework to recognize the given account, pls check https://github.com/hortonworks/cloudbreak/tree/master/authorization-common#how-to-make-an-api-internally-callable

Reply:
@horadla23 since i am not having them declared as internal only they will work. that is the reason i have the @InternalOnly annotation from class level to method level (to the only method that was already present)

Reply:
then others should have also authz related annotation, but as I wrote above, this is an internal API so annotation should stay on class level